pkgs/development/tools/build-managers/waf/default.nix

```diff
@@ -1,44 +1,71 @@
-{ lib, stdenv, fetchFromGitLab, python3, ensureNewerSourcesForZipFilesHook
+{ lib
+, stdenv
+, fetchFromGitLab
+, ensureNewerSourcesForZipFilesHook
+, python3
 # optional list of extra waf tools, e.g. `[ "doxygen" "pytest" ]`
-, withTools ? null
+, extraTools ? []
 }:
-let
-  wafToolsArg = with lib.strings;
-    optionalString (withTools != null) " --tools=\"${concatStringsSep "," withTools}\"";
-in
-stdenv.mkDerivation rec {
+
+stdenv.mkDerivation (finalAttrs: {
   pname = "waf";
   version = "2.0.25";
 
   src = fetchFromGitLab {
     owner = "ita1024";
     repo = "waf";
-    rev = "${pname}-${version}";
-    sha256 = "sha256-wqZEAfGRHhcd7Xm2pQ0FTjZGfuPafRrZAUdpc7ACoEA=";
+    rev = "waf-${finalAttrs.version}";
+    hash = "sha256-wqZEAfGRHhcd7Xm2pQ0FTjZGfuPafRrZAUdpc7ACoEA=";
   };
 
-  nativeBuildInputs = [ python3 ensureNewerSourcesForZipFilesHook ];
+  nativeBuildInputs = [
+    ensureNewerSourcesForZipFilesHook
+    python3
+  ];
+
+  buildInputs = [
+    # waf executable uses `#!/usr/bin/env python`
```

Comment:
So this makes the executables there `patchShebangs` automatically? Please continue the sentence in the comment.

Reply:
No, it just makes Python available at build time.